lib/TPP/Transforms/TileConsumerAndFuseProducers.cpp

```diff
@@ -566,6 +566,19 @@ getTileForEltWiseConsumer(Operation *consumer, Operation *producer,
   return eltWiseTiles;
 }
 
+static bool hasActualOneUse(Value op) {
+  if (op.hasOneUse())
+    return true;
+  llvm::SmallSet<Operation *, 2> users;
+  for (auto &use : op.getUses()) {
+    users.insert(use.getOwner());
+  }
```

Comment:
It should be more efficient to just iterate over users and compare them directly. Sth like:
```
auto users = op.getUsers();
Operation *firstUser = *users.begin();
llvm::all_of(users, [&](Operation *user) { return user == firstUser; });
```